### PR TITLE
Add Watchlist Filters components

### PR DESCRIPTION
**Pull request by tonisevener**
**Phabricator:** 
https://phabricator.wikimedia.org/T342929

### Notes
This is the UI piece to Watchlist Filters. The big new component here is the `WKFormView` (open to a name change here!) and related classes. For now it only has the ability to display a list of single-select or multi-select sections, and updates related view models.

Note this is dependent on https://github.com/wikimedia/wikipedia-ios-data/pull/6 & https://github.com/wikimedia/wikipedia-ios-data/pull/9.

### Test Steps
Load the watchlist filter view. Confirm it themes properly and elements scale with dynamic type. End-to-end testing will need to be done in the client app with an upcoming PR.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ let package = Package(
             targets: ["Components"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/wikimedia/wikipedia-ios-data.git", revision: "40b124b17dd325aa2901054778ce19b8939935b2")
+        .package(url: "https://github.com/wikimedia/wikipedia-ios-data.git", revision: "265a097f2e15eda379bce7045874c96674fbe810")
```
Comment (staykids) [inline Package.swift]: I merged the dependent PR, new head is `a6db372bbc875b36c7e0e3c7e57b8c945dee8720`. Should we update here or on subsequent PR?

File: Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            WKFormView(viewModel: viewModel.formViewModel)
+                .navigationTitle(viewModel.localizedStrings.title)
+                .navigationBarTitleDisplayMode(.inline)
+                .navigationBarItems(
+                    trailing:
+                        Button(action: {
+                            doneAction()
+                        }) {
+                            Text(viewModel.localizedStrings.doneTitle)
+                                .foregroundColor(Color(theme.text))
```
Comment (staykids) [inline Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterView.swift:25]: Pending design Qs here: https://phabricator.wikimedia.org/T342929#9147352

File: Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        let significanceNonMinorEdits: String
+        let userRegistrationHeader: String
+        let userRegistrationUnregistered: String
+        let userRegistrationRegistered: String
+        let typeOfChangeHeader: String
+        let typeOfChangePageEdits: String
+        let typeOfChangePageCreations: String
+        let typeOfChangeCategoryChanges: String
+        let typeOfChangeWikidataEdits: String
+        let typeOfChangeLoggedActions: String
```
Comment (staykids) [inline Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterViewModel.swift:36]: 😅 the things we do for localized strings...

File: Sources/Components/Components/Watchlist/WKWatchlistViewController.swift [hunk 4]
```diff
@@ -37,6 +53,17 @@ public final class WKWatchlistViewController: WKCanvasViewController {
 		addComponent(hostingViewController, pinToEdges: true)
 		self.title = viewModel.localizedStrings.title
 		navigationItem.rightBarButtonItem = filterBarButton
+        viewModel.$activeFilterCount.sink { [weak self] newCount in
```
Comment (staykids) [inline Sources/Components/Components/Watchlist/WKWatchlistViewController.swift:56]: Something odd I've been running into: the selected `WKFormSelectSingleRowView` filters aren't always reflecting their count back into the active filter count. Here's an example of selecting "Not latest revision" and the count not being updated in the watchlist view.

filter_count

File: Tests/ComponentsTests/WKWatchlistFilterViewModelTests.swift [hunk 5]
```diff
@@ -0,0 +1,347 @@
+//
+//  WKWatchlistFilterViewModelTests.swift
+//  
+//
+//  Created by Toni Sevener on 8/11/23.
+//
```
Comment (staykids) [inline Tests/ComponentsTests/WKWatchlistFilterViewModelTests.swift]: Extraneous header.

**Review by staykids — COMMENTED** [review]
So rad! One main note about filter count behavior below.

File: Package.swift (hunk 1 above)
Comment (tonisevener) [reply Package.swift]: I think, as long as the dependent PR in WKData was up-to-date with `main` before you merged it, and `265a097f2e15eda379bce7045874c96674fbe810` was the latest commit in that branch, then this commit will still work. There shouldn't be a need to change it to the `main` PR merge commit here.

File: Sources/Components/Components/Watchlist/WKWatchlistViewController.swift (hunk 4 above)
Comment (tonisevener) [reply Sources/Components/Components/Watchlist/WKWatchlistViewController.swift:56]: @staykids Good catch. I have some confusion with interpreting the Android calls here. I asked here to hopefully clear things up - https://phabricator.wikimedia.org/T342929#9147485.

There's also an unintentional WKData bug that slipped in, I'll put up a fix (https://github.com/wikimedia/wikipedia-ios-data/pull/10) but hold it until I can confirm that this is correct.

**Comment by staykids** [comment]
@tonisevener I'm happy to merge as-is and we can address https://github.com/wikimedia/wikipedia-ios-data/pull/10 and the design questions as answers come in, or leave this up until those can also be addressed in this branch. Just let me know what you prefer, totally up to you!

**Comment by tonisevener** [comment]
@staykids I say let's go ahead and merge this and https://github.com/wikimedia/wikipedia-ios-data/pull/10, and I can put in followup fix PRs as they get answered.

**Pull request merged**